### DISCUSSION TITLE
How to mint an NFT?

### DISCUSSION THREAD

**Discussion by nk-hcode** [wevm/wagmi#2540] [Q&A]
I have write a contract using useContractWrite and provided the contract address, abi and functionName, and when i call write function of this contract write hook, then I want to pass price of that NFT and one more parameter tokenId, I have tried many things but still not working, please let me know what is the proper format for this

**Comment by unefined** [comment]
https://wagmi.sh/examples/contract-write-dynamic
https://wagmi.sh/examples/contract-write
will help you

**Comment by wh173-c47** [comment]
Take care of the RPC spamming, seeing the new construction, I fear most mints will end up with users being rate limited if not correctly managed.... First args sent to usePrepareContractWrite => triggers a call (even if this is just your component mounting any way to avoid this by the way ?)
